xbmc/addons/AddonManager.h

```diff
@@ -104,7 +104,12 @@ namespace ADDON
 
     bool GetInstallableAddons(VECADDONS& addons, const TYPE &type);
 
-    /*! Get the installable addon with the highest version. */
+    /*! \brief Get the installable addon depending on install rules.
+        if checked addon is not installed yet, fall back to the highest version.
+     \param addonId addon to check for update or installation
+     \param addon [out] the retrieved addon pointer - only use if the function returns true.
+     \return true if an addon matching the id is available.
+     */
     bool FindInstallableById(const std::string& addonId, AddonPtr& addon);
```

Comment:
Here in addon manager the doc style is a bit mixed, I prever to use as:
```cpp
    /*! 
     * \brief Get the installable addon depending on install rules.
     *
     * \note If checked addon is not installed yet, it fall back and give the highest version from
     * repositories.
     *
     * \param addonId[in] addon to check for update or installation
     * \param addon[out] the retrieved addon pointer - only use if the function returns true.
     * \return true if an addon matching the id is available.
     */
```
Also was the `if checked addon is not installed yet, fall back to the highest version.` first hard to understand (for me). As I thought the place where call this function should fallback and take on another call from repo.

Reply:
yes it's not quite obvious why we need to fall back. will try to explain it clearer

Reply:
done